### DISCUSSION TITLE
Feedback from the professors

### DISCUSSION THREAD

**Discussion by taeil** [Berkeley-Data/irrigation_detection#57] [Ideas]
From Class chat:     I've added some comments to the paper (attached - let me know if you have issues accessing the comments).
Nice work so far, and this seems like a great class project.
My comments and suggestions are with the aim of this work becoming a conference/workshop submission.
Class project++

At a high level:
1. Currently, the connection with irrigation detection isn't clear throughout the paper: BigEarthNet has an "irrigated land" class, but it looks like your evaluation is on all BigEarthNet classes. 
2. Also, the CA data you explore and evaluate doesn't explicitly address irrigation. Here's some ideas to help with this:
- Perform an evaluation on just the irrigation class of BigEarthNet and report accuracy, i.e. do a binary classification on each image
- What is the size of your CA dataset (how many image chips?)? If you run the binary classifier from ^^^ on the CA dataset, how many irrigated lands does it detect? 
- Can we manually inspect the positives and determine true vs false positive?
- I think this paper could focus more on the unique aspects of remote sensing, 
- or even more specifically, irrigation detection: 
- which augmentations help and why? 
- are all the bands necessary, e.g. rgb vs multi-spectral? 
- Are there augmentations specifically for multispectral data?
- We should make the contributions of the paper explicit at the end of the introduction, e.g. "We present the following contributions in this paper:"
- Let's assume we have a [CA-based] irrigation detection system that can classify image chips with 100% accuracy. 
- What can we do with it? Who would we give it to?
- Small points:
- Ideally, your figures should be in a vector format, e.g. pdfs
- In several places, the paragraphs can be broken into multiple paragraphs (they're a bit long)
- The experiments are missing some details, so one recommendation is to add an appendix and list absolutely everything you can about the settings of each experiment.

**Comment by tchken** [comment]
Re-organize the feedback and assessment, open to discuss to be inlined.

- **Comments from Colorado**

### Issues:

1. BigEarthNet has an "irrigated land" class, but it looks like your evaluation is on all BigEarthNet classes.

**Tasks related - (milestone 1) data processing, replicated supervised model, simClR model.**

2. CA data you explore and evaluate doesn't explicitly address irrigation. 
a. Suggestions - Perform an evaluation on just the irrigation class of BigEarthNet and report accuracy, i.e. do a binary classification on each image.
b. What is the size of your CA dataset (how many image chips?)? If you run the binary classifier from ^^^ on the CA dataset, how many irrigated lands does it detect? Can we manually inspect the positives and determine true vs false positive? 

**Task related -**
**1. (milestone 1) data- download CA dataset with GeoTiff and TFRecord format**
**2. TBD. (next steps after replicated the current  models and work)**

### Suggestions:
1. I think this paper could focus more on the unique aspects of remote sensing, or even more specifically, irrigation detection: which augmentations help and why? are all the bands necessary, e.g. rgb vs multi-spectral? Are there augmentations specifically for multispectral data?

**Task related - (milestone 2) - data augmentations. Deep dive on the bands, features importance.**

2. We should make the contributions of the paper explicit at the end of the introduction, e.g. "We present the following contributions in this paper:"

**Task related - TBD (rephrase the paper, etc)**

3. Let's assume we have a [CA-based] irrigation detection system that can classify image chips with 100% accuracy. What can we do with it? Who would we give it to?

**Task related - TBD (milestone 2 - need to brainstorm)**

4. Paper work - 
a. Ideally, your figures should be in a vector format, e.g. pdfs.
b. In several places, the paragraphs can be broken into multiple paragraphs (they're a bit long).
c. The experiments are missing some details, so one recommendation is to add an appendix and list absolutely everything you can about the settings of each experiment.

**Task related -**
**1. (milestone 1 &2) - update paper with latest results**
**2. (milestone 2) - TBD (rephrase the paper, new findings, etc.)**
**3. (milestone 2) - TBD-(Appendix in the paper)**

**Reply by taeil** [reply]
BigEarthNet has an "irrigated land" class, but it looks like your evaluation is on all BigEarthNet classes.
=> not true. we are doing binary classification. Irrigation (positive) vs all others (negative).

**Reply by tchken** [reply]
From Albetro's email, seems we need to avoid asking similar question when meeting with Colorado.

**Comment by tchken** [comment]
Meeting Notes 2/18/2021
https://docs.google.com/document/d/1kmwdFmAuvI1dGgQPBZetEPtsViI9exK8WLzvfOJfdY4/edit
